pkg/instrumentation/apachehttpd_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrumentation
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+)
+
+func TestInjectApacheHttpdagent(t *testing.T) {
+	tests := []struct {
+		name string
+		v1alpha1.ApacheHttpd
+		pod      corev1.Pod
+		expected corev1.Pod
+	}{
+		{
+			name:        "Clone Container not present",
+			ApacheHttpd: v1alpha1.ApacheHttpd{Image: "foo/bar:1"},
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{},
```

Comment:
I don't think this is a valid pod spec... therefore shouldn't we be unable to inject init containers?

Reply:
It's a test for instrumentation only, checking the result of instrumentation. In that case, it does not need IMHO full application pod spec specified. 
Instrumentation does not validate the application pod/container in any way, I think this is the task for K8S validators?